src/build.h

```diff
@@ -35,6 +35,42 @@ struct Node;
 struct State;
 struct Status;
 
+
+// Set of ready edges, sorted by priority
+class EdgeQueue {
+  struct EdgePriorityCompare {
```

Comment:
Can you add this to the Edge class directly (e.g. Edge::PriorityCompare) and move EdgeQueue definition to `src/graph.h` where it belongs. Also a simple unit-test would be welcome.

Reply:
Moved to `src/graph.h` but also had to move `EdgePriorityCompare` outside the class for sub-classing to work.

Reply:
I meant move it to the Edge class itself, but that ok. Thanks for doing this.